extensions/cornerstone/src/tools/ImageOverlayViewerTool.tsx

```diff
@@ -0,0 +1,258 @@
+import { metaData, StackViewport, Types } from '@cornerstonejs/core';
+import { utilities } from '@cornerstonejs/core';
+import { BaseTool } from '@cornerstonejs/tools';
+import { guid } from '@ohif/core/src/utils';
+
+interface CachedStat {
+  color: number[];
+  overlays: (unknown & {
+    _id: string;
+    pixelDataRaw: ArrayBuffer;
+    type: 'G' | 'R';
+  })[];
+}
+
+/**
+ * Image Overlay Viewer tool is not a traditional tool that requires user interactin.
+ * But it is used to display Pixel Overlays. And it will provide toggling capability.
+ *
+ * The documentation for Overlay Plane Module of DICOM can be found in [C.9.2 of
+ * Part-3 of DICOM standard](https://dicom.nema.org/medical/dicom/2018b/output/chtml/part03/sect_C.9.2.html)
+ */
+class ImageOverlayViewerTool extends BaseTool {
+  static toolName = 'ImageOverlayViewer';
+
+  _renderingViewport: any;
+  _cachedStats: { [key: string]: CachedStat } = {};
+
+  constructor(
+    toolProps = {},
+    defaultToolProps = {
+      supportedInteractionTypes: [],
+      configuration: {
+        showOverlays: true,
+        fillColor: [255, 127, 127, 255],
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  private _getCachedStat(targetId: string, createIfMissing = true): CachedStat {
+    if (!this._cachedStats[targetId] && createIfMissing)
+      this._cachedStats[targetId] = {
+        color: [127, 127, 127, 255], // color (r,g,b,a), default color: gray
+        overlays: [],
+      };
+    return this._cachedStats[targetId];
+  }
+
+  renderAnnotation = (enabledElement, svgDrawingHelper) => {
+    // overlays are toggled off by configuration
+    if (!this.configuration.showOverlays) return false;
+
+    const { viewport } = enabledElement;
+    this._renderingViewport = viewport;
+
+    const imageId = this._getReferencedImageId(viewport);
+    if (!imageId) return;
+
+    const targetId = this.getTargetId(viewport);
+    const cachedStat = this._getCachedStat(targetId);
+
+    let overlays;
```

Comment:
Prefer to use const and assign a new variable rather than filtering the original array.